boards: arm: nrf52840dongle_nrf52840: Use USB CDC as default UART

nordicjm opened the pull request:
Enables use of the USB CDC port as the default UART for this board, as it is a USB dongle and makes more sense to use the on-board USB port instead of a disconnected/floating UART interface.

Fixes #54659
Also fixes https://github.com/mcu-tools/mcuboot/issues/1599

nordicjm committed:
boards: arm: nrf52840dongle_nrf52840: Use USB CDC as default UART

Enables use of the USB CDC port as the default UART for this board,
as it is a USB dongle and makes more sense to use the on-board USB
port instead of a disconnected/floating UART interface.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fabiobaltieri commented on boards/arm/nrf52840dongle_nrf52840/Kconfig.defconfig:
```diff
@@ -24,6 +24,42 @@ config FLASH_LOAD_OFFSET
+config SHELL_BACKEND_SERIAL_CHECK_DTR
+	default SHELL
+
+config USB_DEVICE_REMOTE_WAKEUP
+	default n
```
More of a curiosity: why default this to false? The application only uses it if enabled, it's a cool demo to show that the driver has wakeup support.

nordicjm replied:
Have removed it, might need to reintroduce it if it causes problems (was introduced on thingy53 because if the application does not use the API then it can get into a state whereby the device locks up)

tmon-nordic replied:
Driver supports remote wakeup, but CDC ACM implementation does not seem to issue remote wakeup on its own. Therefore, if host suspends the device (Linux does so if there is no outgoing traffic and no incoming traffic) then the data device wants to write won't be ever picked up (it only will be picked up if host wants to send something).

At this point I would simply continue to use to workaround which is to not advertise remote wakeup (i.e. set `USB_DEVICE_REMOTE_WAKEUP` to `n`).

nordicjm replied:
Have added it back in :)

fabiobaltieri replied:
Interesting, thanks for clarifying. Sounds like there's space for for something like a Kconfig dependency or a conditional `#warning` in the affected class drivers.

nordicjm replied:
Possibly room for improvement in the USB CDC class to add support for it also I assume

nordicjm committed:
doc: release-notes 3.3: Add note on nrf52840dongle UART to USB CDC

Adds a note that the nrf52840dongle_nrf52840 board has switched
from the default console being the unconnected UART to USB CDC.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

stephanosio commented:
Is there any reason why this must go into v3.3? This is not fixing a bug and is an more of an enhancement than a stabilisation change.

nordicjm commented:
The bug is that this board is useless without this change, there is no UART wired up, the reason it was ever left like this goes back to a battle I had to fight strongly for with another board

stephanosio commented:
> The bug is that this board is useless without this change, there is no UART wired up,

I still think this is an enhancement because a user may wire up the UART and use it, or override the defaults to enable CDC and use it as the primary console (i.e. the board is not really _useless_ without this change; it just takes an extra step to use it).

Nevertheless, it is indeed not very sane to have something other than the USB as the default console for a _USB dongle_ and, on second thought, the benefits of merging this and having sane defaults outweigh the risk of breaking something, which I am sure is going to get fixed very promptly without much effort.
